src/dump.c

```diff
@@ -631,6 +640,42 @@ static void jl_serialize_value_(jl_serializer_state *s, jl_value_t *v, int as_li
             jl_serialize_value(s, jl_exprarg(e, i));
         }
     }
+    else if (jl_is_phinode(v)) {
+        jl_array_t *edges = (jl_array_t*)jl_fieldref_noalloc(v, 0);
+        jl_array_t *values = (jl_array_t*)jl_fieldref_noalloc(v, 1);
+        size_t l = jl_array_len(edges);
+        if (l <= 255 && jl_array_len(values) == l) {
```

Comment:
We should just have a special tag for all short 1-d arrays.

Reply:
uh, sure. Not really relevant for this PR though, since this serializer mode is prevented from handle arrays (they aren't an ast node type)